src/lib.rs

```diff
@@ -107,7 +107,7 @@
 //!     match opt {
 //!         Opt::Watch(watch) => {
 //!             log::info!("Starting to watch `cargo check`");
-//!             watch.run(run_command)?;
+//!             watch.run(vec![run_command])?;
```

Comment:
Excellent work!

You see how abstracting the logic behind ShareableChild made everything easier.

Now I would like you to improve the API of the `fn run()` on this watch object. You see, in the past we could provide a command and it would work. Ideally we want to preserve this behavior AND allow passing an iterable over commands. insert why-not-both.gif here

In order to do that, instead of taking the parameter `mut commands: Vec<Command>` you can create a new type and take as parameter `mut commands: impl Into<CommandList>`.

I think I have already said too much so I will leave the rest to you. The goal is to allow this:
- `watch.run(vec![run_command])?;` (list of commands, the easiest)
- `watch.run(run_command)?;` (single command, easy)
- `watch.run(&[run_command])?;` (any iterable actually, you need to get your hand on generic parameters, more complicated, I believe in you)